README.md

```diff
@@ -2,46 +2,157 @@
+
+
+### 6. Setup Oauth 
+Login to [OSM](https://www.openstreetmap.org/) , Click on My Settings and register your local galaxy app to Oauth2applications
+
+![image](https://user-images.githubusercontent.com/36752999/188452619-aababf28-b685-4141-b381-9c25d0367b57.png)
+
+
+Check on read user preferences and Enter redirect URI as following
+```http://127.0.0.1:8000/latest/auth/callback/```
+
+Grab Client ID and Client Secret and put it inside config.txt as OAUTH Block , you can generate secret key for your application by yourself
+
 
-Following this [config sample](https://github.com/hotosm/galaxy-api/blob/master/src/config.txt.sample) setup your database configuration like this config block 
+### 7. Put your credentials inside config.txt
+Insert your config blocks with the database credentials where you have underpass ,insight and tm in your database
 
 ```
-[SOURCE_NAME]
+[INSIGHTS]
 host=localhost
+user=postgres
+password=admin
+database=insights
+port=5432
+
+[UNDERPASS]
+host=localhost
+user=postgres
+password=admin
+database=underpass
+port=5432
+
+[RAW_DATA]
+host=localhost
+user=postgres
+password=admin
+database=raw
+port=5432
+
+[OAUTH]
+client_id= your client id 
+client_secret= your client secret
+url=https://www.openstreetmap.org
+scope=read_prefs
+login_redirect_uri=http://127.0.0.1:8000/latest/auth/callback/
+secret_key=jnfdsjkfndsjkfnsdkjfnskfn
+
+[API_CONFIG]
+env=dev
+
+```
+
+#### Optional Configuration
+
+You can further customize API if you wish with API_CONFIG Block 
+
+```
+[API_CONFIG]
+export_path=exports/ # used to store export path
+api_host=http://127.0.0.1 # you can define this if you have different host 
+api_port=8000
+max_area=100000 # max area to support for rawdata input 
+use_connection_pooling=True # default it will not use connection pooling but you can configure api to use to for psycopg2 connections
+log_level=info #options are info,debug,warning,error
+env=dev # default is dev , supported values are dev and prod
+```
+Based on your requirement you can also customize rawdata exports parameter using EXPORT_UPLOAD block 
+
+```
+[EXPORT_UPLOAD]
+FILE_UPLOAD_METHOD=disk # options are s3,disk , default disk
+AWS_ACCESS_KEY_ID= your id 
+AWS_SECRET_ACCESS_KEY= yourkey 
+BUCKET_NAME= your bucket name 
+```
+##### Setup Tasking Manager Database for TM related development  
+
+You can setup [Tasking manager](https://github.com/hotosm/tasking-manager)  and add those block to config.txt 
+```
+[TM]
+host=
 user=
 password=
-dbname=
 port=
 ```
-
-
-#### 3. Run server
+### 8. Run server
 
 ```uvicorn API.main:app --reload```
 
-#### 4. Navigate to Fast API Docs to get details about API Endpoint 
+### 9. Navigate to Fast API Docs to get details about API Endpoint 
 
 After sucessfully running server , hit [this](http://127.0.0.1:8000/latest/docs) URL on your browser
 
 ```http://127.0.0.1:8000/latest/docs```
 
+### Check Authetication 
+
+1. Hit /auth/login/
+2. Hit Url returned on response 
```

Comment:
I have an error at this step: 
`The requested redirect uri is malformed or doesn't match client redirect URI.`

My generated url:
`https://www.openstreetmap.org/oauth2/authorize/?response_type=code&client_id=Gv5iCf14Yc-hGWkK_TJB-9iNQ7i9zRSX7iRQA44KBfk&redirect_uri=http%3A%2F%2F127.0.0.1%3A8000%2Fv1%2Fauth%2Fcallback%2F&scope=read_prefs&state=iBcdvtQ9ov7yX12wVavBUk81zizES6`

Reply:
this error is generated by osm oauth , because either you have typo error on redirect url on osm oauth setting or on config file of galaxy api , they should match 
on config file login redirect should be like in line95 of readme and it should be same while registering the osm application as well

Reply:
I used the url from the sample config file which has not been updated.
I will do a PR to avoid other people to do the same issue.